e2e/blockstorage/resource_block_storage.go

```diff
@@ -278,3 +343,25 @@ func validateSize(i interface{}, key string) (ws []string, es []error) {
 
 	return
 }
+
+func setPrevState(d *schema.ResourceData, prevVMID, prevName, prevSize interface{}) {
+	d.Set("vm_id", prevVMID)
+	d.Set("name", prevName)
+	d.Set("size", prevSize)
+}
+
+func waitForDetach(apiClient *client.Client, blockStorageID string, project_id int, location string) diag.Diagnostics {
+	for {
+		blockStorage, err := apiClient.GetBlockStorage(blockStorageID, project_id, location)
+		if err != nil {
+			log.Printf("[ERROR] Error getting block storage %s", err)
+			return diag.FromErr(err)
+		}
+		data := blockStorage["data"].(map[string]interface{})
+		if data["status"] == "Available" {
+			break
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
Start adding comments to code so that a reviewer will know why you used this particular line like time.sleep().